### PR TITLE
[SPARK-22324][SQL][PYTHON][FOLLOW-UP] Update setup.py file.

### PR DESCRIPTION
**Pull request by ueshin**
## What changes were proposed in this pull request?

This is a follow-up pr of #19884 updating setup.py file to add pyarrow dependency.

## How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by ueshin** [comment]
Btw, should we add `'Programming Language :: Python :: 3.6'` to `classifiers`?

**Comment by HyukjinKwon** [comment]
Yea, I think we could. I added the support and tested it before - SPARK-19019. I think it's okay to add it they are just metadata AFAIK.

**Comment by ueshin** [comment]
@HyukjinKwon Thanks! I'll add it soon.

**Review by HyukjinKwon — APPROVED** [review]
LGTM

Not a big deal but I know one more place we might also update - https://github.com/apache/spark/blob/master/python/README.md#python-requirements

**Comment by ueshin** [comment]
@HyukjinKwon I'll update it as well.

File: python/README.md [hunk 1]
```diff
@@ -29,4 +29,4 @@ The Python packaging for Spark is not intended to replace all of the other use c
 
 ## Python Requirements
 
-At its core PySpark depends on Py4J (currently version 0.10.6), but additional sub-packages have their own requirements (including numpy and pandas).
+At its core PySpark depends on Py4J (currently version 0.10.6), but additional sub-packages have their own requirements (including numpy, pandas, and pyarrow).
```
Comment (viirya) [inline python/README.md]: This sounds like mandatory, but I think pyarrow is still an optional choice. Right?

File: python/setup.py [hunk 2]
```diff
@@ -201,7 +201,7 @@ def _supports_symlinks():
         extras_require={
             'ml': ['numpy>=1.7'],
             'mllib': ['numpy>=1.7'],
-            'sql': ['pandas>=0.19.2']
+            'sql': ['pandas>=0.19.2', 'pyarrow>=0.8.0']
```
Comment (viirya) [inline python/setup.py]: If no pyarrow is installed, will setup force users to install it?

Comment (HyukjinKwon) [reply python/setup.py]: Nope, `extras_require` does not do anything in normal cases but they can be installed together with a dev option via pip IIRC.

File: python/README.md (hunk 1 above)
Comment (HyukjinKwon) [reply python/README.md]: Yea, Pandas and PyArrow are optional. Maybe, it's nicer if we have some more details here too.

Comment (ueshin) [reply python/README.md]: I added some more details. WDYT?

File: python/README.md [hunk 3]
```diff
@@ -29,4 +29,4 @@ The Python packaging for Spark is not intended to replace all of the other use c
 
 ## Python Requirements
 
-At its core PySpark depends on Py4J (currently version 0.10.6), but additional sub-packages have their own requirements (including numpy and pandas).
+At its core PySpark depends on Py4J (currently version 0.10.6), but additional sub-packages might have their own requirements declared as "Extras" (including numpy, pandas, and pyarrow). You can install the requirements by specifying their extra names.
```
Comment (HyukjinKwon) [inline python/README.md]: Ah, I see. How about simply:

```
At its core PySpark depends on Py4J (currently version 0.10.6), but some additional sub-packages have their own 
extra requirements for some features (including numpy, pandas, and pyarrow).
```

for now? I just noticed we are a bit unclear on this (e.g., actually I have been under impression that NumPy is required for ML/MLlib so far) but I think this roughly describes it correctly and is good enough.

Will maybe try to make a followup to fully describe it later. This PR targets PyArrow anyway.

Comment (HyukjinKwon) [reply python/README.md]: Not a big deal anyway. I am actually fine as is too if you prefer @ueshin.

**Comment by HyukjinKwon** [comment]
Still LGTM

File: python/README.md (hunk 3 above)
Comment (ueshin) [reply python/README.md]: Let's use the simple one you suggested and leave the detailed description for the future prs.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**